arrow.go

```diff
@@ -125,45 +217,28 @@ func (a *Arrow) QueryContext(ctx context.Context, query string, args ...any) (ar
 	if err != nil {
 		return nil, err
 	}
-	defer stmt.Close()
```

Comment:
Do we need to move this down into the `err != nil` calls? If we `defer` here, we'll clean up there, no?

Reply:
yes, your are right!